RocketApp/Features/Sources/RocketDetail/RocketDetailView.swift

```diff
@@ -1,26 +1,171 @@
 import ComposableArchitecture
+import RocketLaunch
 import RocketsClient
 import SwiftUI
+import UIToolkit
 
 public struct RocketDetailView: View {
   var store: StoreOf<RocketDetailCore>
+  @ObservedObject var viewStore: ViewStoreOf<RocketDetailCore>
 
   public init(store: StoreOf<RocketDetailCore>) {
     self.store = store
+    self.viewStore = ViewStore(store)
   }
 
   public var body: some View {
-    WithViewStore(self.store) { viewStore in
-      VStack {
-        Text(viewStore.rocketData.id)
+    ScrollView(showsIndicators: false) {
+      VStack(alignment: .leading) {
+        section(.overview) {
+          Text(viewStore.rocketData.overview)
+            .font(.body)
+        }
+        .padding(.bottom)
 
-        Text(viewStore.rocketData.name)
+        section(.parameters) {
+          VStack {
+            parameters
+            // TODO: All view logic move to ViewState.
+            Toggle(
+              viewStore.isUsMetrics ? .usMetrics : .euMetrics,
+              isOn: viewStore.binding(get: \.isUsMetrics, send: RocketDetailCore.Action.setToUSMetrics)
+            )
+            .padding(.horizontal)
+          }
+        }
+
+        section(.firstStage) {
+          stageList(items: firstStageItems)
+        }
+        .padding()
+
+        section(.secondStage) {
+          stageList(items: secondStageItems)
+        }
+        .padding()
+
+        section(.photos) {
+          VStack {
+            ForEach(viewStore.rocketData.photos, id: \.self) { path in
+              AsyncImage(
+                url: URL(string: path),
+                content: { $0.resizable().scaledToFit().cornerRadius(24) },
+                placeholder: { EmptyView() }
+              )
+            }
+          }
+        }
       }
     }
+    .padding()
+    .navigationTitle(viewStore.rocketData.name)
+    .navigationBarItems(trailing: Button(.launch) { viewStore.send(.rocketLaunchTapped) })
+    .navigationDestination(
+      isPresented: viewStore.binding(
+        get: { $0.route != nil },
+        send: RocketDetailCore.Action.setNavigation(isActive:)
+      )
+    ) { destination }
+  }
+
+  @ViewBuilder
+  private var destination: some View {
+    IfLetStore(store.scope(state: \.rocketLaunchState, action: RocketDetailCore.Action.rocketLaunch)) {
+      RocketLaunchView(store: $0)
+    }
+  }
+
+  @ViewBuilder
+  private func section<V: View>(_ caption: LocalizedStringKey, _ content: @escaping () -> V) -> some View {
+    VStack(alignment: .leading, spacing: 8) {
+      Text(caption)
+        .bold()
+        .font(.body)
+
+      content()
+    }
+  }
+
+  private var parameters: some View {
+    HStack(spacing: 16) {
+      paramWindow(type: .height)
+
+      paramWindow(type: .diameter)
+
+      paramWindow(type: .mass)
+    }
+  }
+
+  private func paramWindow(type: RocketDetail.RocketParameters, backgroundColor: Color = .pink) -> some View {
```

Comment:
Jako nech to tu, na to nemame cas to predelavat. Ale takhle by se SwiftUI moc delat nemelo. To se ma delat tak, ze delas maly views, s vlastnim previews a ty skladas do sebe. Ne ze si delas privatni funkce, co ti views generujou...